src/h5web/providers/jupyter/utils.ts

```diff
@@ -113,16 +103,11 @@ export function convertDtype(dtype: string): HDF5Type {
       return {
         class: HDF5TypeClass.String,
         charSet: 'ASCII',
-        length: length || 'H5T_VARIABLE',
+        length,
```

Comment:
`|S` without a number is not possible, so I'm passing the `length` without a fallback.